pkg/addons2/adminconsole/upgrade.go

```diff
@@ -5,34 +5,35 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/replicatedhq/embedded-cluster/pkg/helm"
-	"github.com/replicatedhq/embedded-cluster/pkg/runtimeconfig"
-	"github.com/replicatedhq/embedded-cluster/pkg/versions"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (a *AdminConsole) Upgrade(ctx context.Context, kcli client.Client) error {
-	if err := a.prepare(); err != nil {
-		return errors.Wrap(err, "prepare admin console")
+func (a *AdminConsole) Upgrade(ctx context.Context, kcli client.Client, hcli helm.Client, overrides []string) error {
+	exists, err := hcli.ReleaseExists(ctx, namespace, releaseName)
+	if err != nil {
+		return errors.Wrap(err, "check if release exists")
+	}
+	if !exists {
+		// admin console must exist during upgrade
+		return errors.New("admin console release not found")
 	}
 
-	hcli, err := helm.NewHelm(helm.HelmOptions{
-		KubeConfig: runtimeconfig.PathToKubeConfig(),
-		K0sVersion: versions.K0sVersion,
-	})
+	values, err := a.GenerateHelmValues(ctx, kcli, overrides)
 	if err != nil {
-		return errors.Wrap(err, "create helm client")
+		return errors.Wrap(err, "generate helm values")
 	}
 
 	_, err = hcli.Upgrade(ctx, helm.UpgradeOptions{
 		ReleaseName:  releaseName,
 		ChartPath:    Metadata.Location,
 		ChartVersion: Metadata.Version,
-		Values:       helmValues,
+		Values:       values,
 		Namespace:    namespace,
+		Labels:       getBackupLabels(),
 		Force:        false,
 	})
 	if err != nil {
-		return errors.Wrap(err, "upgrade admin console")
+		return errors.Wrap(err, "upgrade")
```

Comment:
```suggestion
		return errors.Wrap(err, "helm upgrade")
```

Reply:
i feel like "upgrade admin console" is more accurate here though

Reply:
i dont think so because that will be duplicated if you look up the stack

Reply:
looks like you comment was on old code. it only says `upgrade` in the final version

Reply:
https://github.com/replicatedhq/embedded-cluster/pull/1757#discussion_r1942014205